src/main/scala/lasp/hapi/service/latis2/Latis2Interpreter.scala

```diff
@@ -61,4 +67,32 @@ class Latis2Interpreter[F[_]: Sync](dir: String) extends HapiInterpreter[F] {
       // Get the metadata.
       metadata <- EitherT.liftF(Latis2Util.getDatasetMetadata(dataset))
     } yield metadata
+
+  override def getData(req: DataRequest): F[dm.Dataset] = req match {
+    case DataRequest(id, minTime, maxTime, params, _, _) =>
+      val ops = List(
+        Selection(s"time>=${minTime.format(Time.format)}"),
+        Selection(s"time<${maxTime.format(Time.format)}")
+      ) ++ params.fold(
+        List.empty[Operation]
+      )(
+        ps => List(Projection("time" :: ps.toList))
+      )
+      Latis2Util.getDataset(id, ops)
+  }
+
+  override def writeData(data: dm.Dataset): Stream[F, String] =
+    Stream.eval(data.pure[F]).flatMap {
+      case dm.Dataset(Function(it)) =>
+        val rows = it.map {
```

Comment:
Since `Sample` in v2 is a Tuple, I wonder if you could simply flatten it, exclude the `Index`, and `toString` everything.